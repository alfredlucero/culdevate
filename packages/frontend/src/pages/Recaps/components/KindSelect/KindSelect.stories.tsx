import React from "react";
import { storiesOf } from "@storybook/react";
import KindSelect from "./index";
import { RecapKind } from "../../recaps.interface";

storiesOf("RecapsPage/KindSelect", module)
  .add("Option Selected", () => (
    <div style={{ width: "400px", padding: "30px" }}>
      <KindSelect id="kindSelect" value={RecapKind.WorkExperience} onChange={() => {}} />
    </div>
  ))
  .add("Not selected", () => (
    <div style={{ width: "400px", padding: "30px" }}>
      <KindSelect id="kindSelect" value="" onChange={() => {}} />
    </div>
  ));
