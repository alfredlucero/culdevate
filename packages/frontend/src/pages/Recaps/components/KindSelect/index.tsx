import React from "react";
import Select, { SelectProps, SelectOption } from "../../../../components/Select";
import { RecapKind } from "../../recaps.interface";

interface KindSelectProps extends Omit<SelectProps, "label" | "options" | "placeholder" | "required"> {
  value: RecapKind | "";
}

const recapKindOptionMap = {
  ...RecapKind,
};

const kindOptions: SelectOption[] = Object.entries(recapKindOptionMap).map(([kindKey, kindValue]) => ({
  label: kindValue,
  value: kindValue,
}));

const KindSelect: React.FC<KindSelectProps> = selectProps => {
  return (
    <Select
      label="What kind of recap is it?"
      options={kindOptions}
      placeholder="Select one kind of recap..."
      required={true}
      {...selectProps}
    />
  );
};

export default KindSelect;
