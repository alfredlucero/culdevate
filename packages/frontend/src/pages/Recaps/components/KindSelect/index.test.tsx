import React from "react";
import { render } from "@testing-library/react";
import KindSelect from "./index";

describe("<KindSelect />", () => {
  test("should render without error", () => {
    const { container } = render(
      <KindSelect
        id="kindSelect"
        value="Work Experience"
        onChange={() => {}}
        data-testid="Kind Select"
        className="extra-kind-select-class"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
