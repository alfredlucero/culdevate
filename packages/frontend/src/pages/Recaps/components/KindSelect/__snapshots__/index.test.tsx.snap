// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KindSelect /> should render without error 1`] = `
<div>
  <div
    class="extra-kind-select-class"
  >
    <label
      class="block text-gray-700 text-sm font-bold mb-2"
      for="kindSelect"
    >
      What kind of recap is it?
       
      
    </label>
    <div
      class="inline-block relative"
    >
      <select
        aria-required="true"
        class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
        data-testid="kind-select-id"
        id="kindSelect"
        name="kindSelect"
        required=""
      >
        <option
          value=""
        >
          Select one kind of recap...
        </option>
        <option
          value="Work Experience"
        >
          Work Experience
        </option>
        <option
          value="Education"
        >
          Education
        </option>
        <option
          value="Accomplishments"
        >
          Accomplishments
        </option>
        <option
          value="Publications"
        >
          Publications
        </option>
        <option
          value="Skills"
        >
          Skills
        </option>
        <option
          value="Side Projects"
        >
          Side Projects
        </option>
        <option
          value="Organizations"
        >
          Organizations
        </option>
        <option
          value="References"
        >
          References
        </option>
        <option
          value="Other"
        >
          Other
        </option>
      </select>
      <div
        class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
      >
        <svg
          class="fill-current h-4 w-4"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
